py3-oauthenticator.yaml

```diff
@@ -38,6 +38,17 @@ pipeline:
 
   - uses: strip
 
+test:
+  environment:
+    contents:
+      packages:
+        - openssl
```

Comment:
This is suspicious. If we need these to run the test, shouldn't they be runtime dependencies of the package?

Reply:
Looks like it was built as part of a larger set of K8s related packages. I'll add it as a runtime dep as without it will throw errors.

Reply:
Is this a real dependency?  Python already brings in libssl it seems unlikely to me that this module actually needs specifically openssl as a dep?

Reply:
This is what I get without it during an import:

```
2024/10/11 11:42:28 INFO using python /usr/bin/python3.12 uses=python/import
2024/10/11 11:42:28 WARN Traceback (most recent call last): uses=python/import
2024/10/11 11:42:28 WARN   File "<string>", line 1, in <module> uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/oauthenticator/__init__.py", line 4, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from .bitbucket import *  # noqa uses=python/import
2024/10/11 11:42:28 WARN     ^^^^^^^^^^^^^^^^^^^^^^^^ uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/oauthenticator/bitbucket.py", line 7, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from jupyterhub.auth import LocalAuthenticator uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/jupyterhub/auth.py", line 26, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from .handlers.login import LoginHandler uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/jupyterhub/handlers/__init__.py", line 1, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from . import base, login, metrics, pages uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/jupyterhub/handlers/base.py", line 48, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from ..user import User uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/jupyterhub/user.py", line 16, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from .crypto import CryptKeeper, EncryptionUnavailable, InvalidToken, decrypt, encrypt uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/jupyterhub/crypto.py", line 12, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from cryptography.fernet import Fernet, InvalidToken, MultiFernet uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/cryptography/fernet.py", line 14, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from cryptography.exceptions import InvalidSignature uses=python/import
2024/10/11 11:42:28 WARN   File "/usr/lib/python3.12/site-packages/cryptography/exceptions.py", line 9, in <module> uses=python/import
2024/10/11 11:42:28 WARN     from cryptography.hazmat.bindings._rust import exceptions as rust_exceptions uses=python/import
2024/10/11 11:42:28 WARN RuntimeError: OpenSSL 3.0's legacy provider failed to load. This is a fatal error by default, but cryptography supports running without legacy algorithms by setting the environment variable CRYPTOGRAPHY_OPENSSL_NO_LEGACY. If you did not expect this error, you have likely made a mistake with your OpenSSL configuration. uses=python/import
```

Reply:
huh weird.

Reply:
> This is suspicious. If we need these to run the test, shouldn't they be runtime dependencies of the package?

I don't think so - essentially this is our way (today) of loading python packages, instead of pip. i.e a pip install X wouldn't add openssl. Also if we add as a runtime dep, if someone was running different openssl versions on their image, it may cause a conflict

Reply:
If they have a different version of openssl and it has a `provides=openssl` the runtime dependency will still be satisfied. 

As it stands this package is not functional without an openssl dependency.